goutubedl.go

```diff
@@ -488,12 +517,18 @@ type DownloadOptions struct {
 	PlaylistIndex int
 }
 
-func (result Result) DownloadWithOptions(ctx context.Context, options DownloadOptions) (*DownloadResult, error) {
+func (result Result) DownloadWithOptions(
+	ctx context.Context,
+	options DownloadOptions,
+) (*DownloadResult, error) {
 	debugLog := result.Options.DebugLog
 
 	if !result.Options.noInfoDownload {
-		if (result.Info.Type == "playlist" || result.Info.Type == "multi_video") && options.PlaylistIndex == 0 {
-			return nil, fmt.Errorf("can't download a playlist when the playlist index options is not set")
+		if (result.Info.Type == "playlist" || result.Info.Type == "multi_video") &&
```

Comment:
Should channel also fail? now i don't remember what `PlaylistIndex == 0` means

Reply:
Upstream has this to say about `--playlist-items`/`PlaylistIndex`:

> Comma separated playlist_index of the items to download

https://github.com/yt-dlp/yt-dlp/blob/998dffb5a2343ec709b3d6bbf2bf019649080239/yt_dlp/options.py#L573-L580

In our case, we just want an `int` which presumably gets cast into a `string` arg when running the actual ytdl command.

Technically `DownloadOptions.PlaylistIndex` should be a `string` or, perhaps more correctly, `[]int`. But this type change is better handled in a separate PR, imo.

Aside from that, I agree that channel should also fail, which I've just addressed: https://github.com/wader/goutubedl/pull/172/commits/3f0fe87c2e2d8cd3709fc205e4fee8adbc9d9db3

Reply:
I see, now i remember more. Should it always return error when trying to download a channel even if playlist index is set? or is "index" into a channel to a entry possible somehow with yt-dlp? sorry was a while since i messed around with yt-dlp cli

Reply:
Yeah it seems like `yt-dlp` is smart enough to apply `--playlist-items` to a channel. I was worried they would return just the first playlist in its entirety (since playlists are the top level item in a channel), but they do descend first before performing the indexing:

```shell
yt-dlp -J  --playlist-items 1 https://www.youtube.com/channel/UCHDm-DKoMyJxKVgwGmuTaQA | jq . | grep webpage_url
          "webpage_url": "https://www.youtube.com/watch?v=mpuHeebTk_0",
          "webpage_url_basename": "watch",
          "webpage_url_domain": "youtube.com",
      "webpage_url": "https://www.youtube.com/channel/UCHDm-DKoMyJxKVgwGmuTaQA/videos",
  "webpage_url": "https://www.youtube.com/channel/UCHDm-DKoMyJxKVgwGmuTaQA",
  "webpage_url_basename": "UCHDm-DKoMyJxKVgwGmuTaQA",
  "webpage_url_domain": "youtube.com",
```